packaging/debian-sid/snapd.postrm

```diff
@@ -96,6 +96,10 @@ if [ "$1" = "purge" ]; then
         rm -f "/etc/systemd/system/multi-user.target.wants/$unit"
     done
 
+    # now stop snapd itself to ensure we can clearly remove all snapd data
```

Comment:
would it be helpful/more correct to have a loop like this at the top of the `if [ "$1" = "purge" ];` we should just ask snapd to remove all the snaps first like (psuedo shell code):
```bash
for sn in $(snap list | awk '{print $1}' | grep "only the non-base snaps"); do
  case "$sn" in 
    core|snapd)
      # skip til the end
    ;;
    *)
      snap remove $sn || true 
    ;;
  esac
done
for sn in $(snap list | awk '{print $1}' | grep "only the base snaps"); do
  snap remove $sn || true
done
```

this way we let snapd do its absolute best to remove snaps "properly", then we proceed with tearing out the world underneath snapd by deleting the systemd units and umounting the snap files.

Reply:
This can be done in prerm only, and assuming one can actually talk to snapd.